readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -120,8 +218,10 @@ def test_incremental_state_update_with_no_update(self):
                     'output': u'',
                     'state': BUILD_STATE_CLONING,
                     'builder': mock.ANY,
-                    'exit_code': 0,
                 })
+            self.assertIsNone(build_env.failure)
+        # No commands were saved
+        self.assertFalse(self.mocks.mocks['api_v2.build'].post.called)
```

Comment:
I think there is a problem here.

This method _should have been called_ (`api_v2.build`) since it's the one that has updated our builds inside the `with` context.

I noticed that maybe you missed to add the `()` to the mocks. Take a look a the other lines that use the same call:

    self.mocks.mocks['api_v2.build']().put.assert_called_with({

Reply:
Of what I remember from that long afternoon of debugging, the put method was used to save a build. I going to take a more closer look to the code.